lib/mollie/api/object/payment.rb

```diff
@@ -15,50 +15,103 @@ class Payment < Base
 
         attr_accessor :id,
-                      :status,
                       :mode,
-                      :amount,
-                      :description,
-                      :method,
                       :created_datetime,
+                      :status,
                       :paid_datetime,
-                      :expired_datetime,
                       :cancelled_datetime,
+                      :expired_datetime,
+                      :expiry_period,
+                      :amount,
+                      :amount_refunded,
+                      :amount_remaining,
+                      :description,
+                      :method,
+                      :metadata,
+                      :locale,
+                      :profile_id,
+                      :settlement_id,
                       :customer_id,
                       :recurring_type,
                       :mandate_id,
                       :subscription_id,
-                      :settlement_id,
-                      :metadata,
-                      :details,
-                      :links
+                      :links,
+                      :details
 
         def open?
-          @status == STATUS_OPEN
+          status == STATUS_OPEN
         end
 
         def cancelled?
-          @status == STATUS_CANCELLED
+          status == STATUS_CANCELLED
         end
 
         def expired?
-          @status == STATUS_EXPIRED
+          status == STATUS_EXPIRED
         end
 
         def paid?
-          !@paid_datetime.nil?
+          !!paid_datetime
         end
 
         def paidout?
-          @status == STATUS_PAIDOUT
+          status == STATUS_PAIDOUT
         end
 
         def refunded?
-          @status == STATUS_REFUNDED
+          status == STATUS_REFUNDED
+        end
+
+        def details=(details)
+          @details = OpenStruct.new(details) if details.is_a?(Hash)
+        end
+
+        def created_datetime=(created_datetime)
+          @created_datetime = Time.parse(created_datetime.to_s) rescue nil
+        end
+
+        def paid_datetime=(paid_datetime)
+          @paid_datetime = Time.parse(paid_datetime.to_s) rescue nil
+        end
+
+        def cancelled_datetime=(cancelled_datetime)
+          @cancelled_datetime = Time.parse(cancelled_datetime.to_s) rescue nil
+        end
+
+        def expired_datetime=(expired_datetime)
+          @expired_datetime = Time.parse(expired_datetime.to_s) rescue nil
+        end
+
+        def amount=(amount)
+          @amount = BigDecimal.new(amount) if amount
+        end
+
+        def amount_remaining=(amount_remaining)
+          @amount_remaining = BigDecimal.new(amount_remaining) if amount_remaining
+        end
+
+        def amount_refunded=(amount_refunded)
+          @amount_refunded = BigDecimal.new(amount_refunded) if amount_refunded
         end
 
         def payment_url
-          @links && @links.payment_url
+          links && links['payment_url']
```

Comment:
There was an issue with the old code (see issue #4) where it would produce an error when no `payment_url` exists on the `links` object:

```
(undefined method `paymentUrl' for #<Mollie::API::Object::Base:0x007f4c2c1c4498>):
  mollie-api-ruby (1.1.1) lib/Mollie/API/Object/Payment.rb:34:in `getPaymentUrl'
```

Is this still an issue with this new code?